File: mobile/src/main/java/net/maiatoday/levelbest/data/Entry.java
package net.maiatoday.levelbest.data;

import java.util.Date;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.Required;

/**
 * Created by maia on 2017/02/04.
 */

public class Entry extends RealmObject {
    @Required
    @PrimaryKey
    String id;
    Date when;
    String where;
    RealmList<Tag> tags;
    RealmList<Mood> sentiments;
    int sentiment;
    int energyLevel;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Date getWhen() {
        return when;
    }

    public void setWhen(Date when) {
        this.when = when;
    }

    public String getWhere() {
        return where;
    }

    public void setWhere(String where) {
        this.where = where;
    }

    public RealmList<Tag> getTags() {
        return tags;
    }

    public void setTags(RealmList<Tag> tags) {
        this.tags = tags;
    }

    public int getSentiment() {
        return sentiment;
    }

    public void setSentiment(int sentiment) {
        this.sentiment = sentiment;
    }

    public int getEnergyLevel() {
        return energyLevel;
    }

    public void setEnergyLevel(int energyLevel) {
        this.energyLevel = energyLevel;
    }
}

File: mobile/src/main/kotlin/net/maiatoday/levelbest/data/Entry.kt
package net.maiatoday.levelbest.data

import java.util.Date

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required

/**
 * Mood/Energy data point entry
 * Created by maia on 2017/02/04.
 */

open class Entry(
        @Required @PrimaryKey open var id: String = "",
        open var timeStamp: Date = Date(),
        open var where: String = "",
        open var tags: RealmList<Tag> = RealmList(),
        open var moods: RealmList<Mood> = RealmList(),
        open var sentiment: Int = 0,
        open var energyLevel: Int = 0
) : RealmObject() {
    // Realm overloads generated getters and setters
}
